Automattic/wp-calypso
# A/B Test Adding G Suite SKUs Select to CTA

#### Changes proposed in this Pull Request

* Add Business Option to G Suite Purchase CTA AB Test
* Add new `/email/:domain/gsuite/new/:planType/:siteSlug` path
* Move `/email/:domain/gsuite-add-users/:siteSlug` to `/email/:domain/gsuite/add-users/:siteSlug` 
* Refactoring the G Suite Purchase CTA
* Added Tracks events for viewing the CTA and clicking the "Get G Suite X" button
* Removed G Suite Purchase CTA Snapshot tests now that component has increased connection to Redux state

**Wide**
<img width="1056" alt="Wide Example" src="https://user-images.githubusercontent.com/2810519/55606234-dd085880-572c-11e9-912f-07e50a9c5293.png">

**Mobile**
<img width="450" alt="Mobile Example" src="https://user-images.githubusercontent.com/2810519/55659359-2ea8f580-57b6-11e9-9a4b-fddd0b8c384e.png">

**"Unlimited" Fine Print**
<img width="450" alt="Screen Shot 2019-04-08 at 11 47 01 AM" src="https://user-images.githubusercontent.com/2810519/55748662-12929780-59f4-11e9-9d8d-b0e5ceb7dafb.png">


#### Testing instructions

1. Enable Analytics logging by typing `localStorage.setItem( 'debug', 'calypso:analytics:*' )` into the console of your browser dev tools
1. Open the "Bug" Menu and confirm you are in the "hide" group of the `gSuiteSkuSelect` AB Test <img width="511" alt="Screen Shot 2019-04-05 at 2 50 29 PM" src="https://user-images.githubusercontent.com/2810519/55658445-8a717f80-57b2-11e9-9e47-02a6a3fffc1a.png">
1. Navigate to `/email/:domain/manage/:siteSlug` for a domain and site without G Suite
1. Confirm what you see matches this screen shot: <img width="969" alt="Screen Shot 2019-04-05 at 2 55 18 PM" src="https://user-images.githubusercontent.com/2810519/55658567-f227ca80-57b2-11e9-8db1-64ae53baa0b6.png">
1. Make sure your Dev Tools are open so you can confirm Tracks events are correct ( clearing it would probably help )
1. Click the "Add G Suite" Button
1. Confirm you have been taken to the `Add G Suite` Page: <img width="733" alt="Screen Shot 2019-04-05 at 2 58 37 PM" src="https://user-images.githubusercontent.com/2810519/55658665-53e83480-57b3-11e9-8855-06557c950440.png">
1. Inspect the url and confirm it matches `/email/:domain/gsuite/new/basic/:siteSlug`
1. Check your Dev tools console and verify a `calypso_email_gsuite_purchase_cta_get_gsuite_click` event was recorded, with the properties `plan_type: 'basic'` and `domain` for the domain
1. Add G Suite user info and continue to the cart
1. Confirm that the item in the cart is regular "G Suite"
1. Don't complete the purchase, remove all items from cart
1. Switch into the "show" group of the `gSuiteSkuSelect` AB Test
1. Navigate back to `/email/:domain/manage/:siteSlug`
1.  Confirm what you see matches this screen shot: <img width="974" alt="Screen Shot 2019-04-05 at 3 07 27 PM" src="https://user-images.githubusercontent.com/2810519/55658998-92322380-57b4-11e9-9350-dcf54501295a.png">
1. Click the "Add G Suite Business" Button
1. Confirm you have been taken to the `Add G Suite` Page
1. Inspect the url and confirm it matches `/email/:domain/gsuite/new/business/:siteSlug`
1. Check your Dev tools console and verify a `calypso_email_gsuite_purchase_cta_get_gsuite_click` event was recorded, with the properties `plan_type: 'business'` and `domain` for the domain
1. Add G Suite user info and continue to the cart
1. Confirm that the item in the cart is "G Suite with Unlimited Storage" and the price is correct
1. Don't complete the purchase, remove all items from cart
1. Navigate to `/email/:domain/manage/:siteSlug` for a site & domain with G Suite
1. Click the "Add G Suite User" button
1. Confirm you have been taken to the `Add G Suite` Page
1. Inspect the url and confirm it matches `/email/:domain/gsuite/add-users/:siteSlug`
1. Change the url to `/email/:domain/add-gsuite-users/:siteSlug`
1. Confirm you are redirected to the same page
1. Add G Suite User info and continue to cart
1. Confirm the item in the cart is "Extra License for G Suite"
